src/gui/qgssnapindicator.cpp

```diff
@@ -0,0 +1,93 @@
+/***************************************************************************
+  qgssnapindicator.cpp
+  --------------------------------------
+  Date                 : October 2017
+  Copyright            : (C) 2017 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgssnapindicator.h"
+
+#include "qgsmapcanvas.h"
+#include "qgssettings.h"
+#include "qgsvectorlayer.h"
+#include "qgsvertexmarker.h"
+
+#include <QToolTip>
+
+
+QgsSnapIndicator::QgsSnapIndicator( QgsMapCanvas *canvas )
+  : mCanvas( canvas )
+{
+}
+
+QgsSnapIndicator::~QgsSnapIndicator()
+{
+}
+
+void QgsSnapIndicator::setMatch( const QgsPointLocator::Match &match )
+{
+  mMatch = match;
+
+  if ( !mMatch.isValid() )
+  {
+    mSnappingMarker.reset();
+    QToolTip::hideText();
+  }
+  else
+  {
+    if ( !mSnappingMarker )
+    {
+      mSnappingMarker.reset( new QgsVertexMarker( mCanvas ) );
+      mSnappingMarker->setPenWidth( 3 );
+    }
+
+    QgsSettings s;
+
+    QColor color = s.value( QStringLiteral( "/qgis/digitizing/snap_color" ), QColor( Qt::magenta ) ).value<QColor>();
+    mSnappingMarker->setColor( color );
+
+    int iconType;
+    if ( match.hasVertex() )
+    {
+      if ( match.layer() )
+        iconType = QgsVertexMarker::ICON_CROSS;  // vertex snap
```

Comment:
I suggest to set new AutoCAD like icons in QgsVertexMarker (see my [proposal](https://github.com/qgis/QGIS-Enhancement-Proposals/issues/107#issuecomment-339251458))
Also, to respect the "standard" with CAD tools, it could be ICON_BOX

Reply:
So if I understand correctly, you suggest ICON_BOX for vertex snaps and a new ICON_DOUBLE_TRIANGLE for segment snaps (in the comment below)?

Reply:
Yes

Reply:
But the best is to leave the choice to the users: select the icon in a combox for example.

Reply:
Is there any particular reason why you would like "box" marker for vertex snaps and "double triangle" marker for segment snaps? I quite like the idea to keep "cross" marker for vertices because it is the most common case and "cross" is more intuitive (to me - as a non-CAD user).

Regarding choice - I would prefer not to have marker types customizable by users. There are already too many options that users can tweak...

Reply:
Very honestly the habit of using AutoCAD and the QAD plugin.
I agree that they are not the most beautiful icons in the world, but you know what it is when you have your little habits...:)

Maybe we aren't a lot, on the QGis github, coming from the CAD world, but I sincerely think it's easier for a QGis user (who until now had only one choice in the marker) to change his markers, rather than potential users making the transition from CAD to GIS. Don't we underestimate these little habits?

I won't go so far as to say that this is a "de facto"standard, but I think we break more habits by proposing markers used in another context rather than aligning with the tools already used, including in QGis (QAD). I also just checked, on ArcGis, it's also a box when snapping on a vertex. The other markers are also close to AutoCAD.
ArcGis toolbar:
![image](https://user-images.githubusercontent.com/7521540/32075559-75369208-ba9d-11e7-93f5-88e5b2adc7fa.png)

I totally understand that others have a different approach. That's why I suggest that you leave the choice to the user in these markers. It is then up to us to propose settings to align ourselves with other tools (AutoCAD/QAD/ArcGis, QCad which use only circle, new QGis standard?).  This may be new options, but IMHO they can be managed through the profile tools.

Reply:
With the latest commit I have switched vertex snap to box and edge snap to double triangle.

For customization of marker types I strongly feel again allowing that - other systems seem to have them fixed as well and I think it is good for consistency.

Reply:
Thank you martin.